test/index.test.ts

```diff
@@ -208,6 +209,103 @@ describe('PaymentSocket', function () {
     })
   })
 
+  describe('pay', function () {
+    it('should lower the maximum balance and resolve when the balance is below the maximum', async function () {
+      await this.clientSocket.pay(2000)
+      assert.equal(this.clientSocket.balance, '-2000')
+
+      await this.clientSocket.pay(1500)
+      assert.equal(this.clientSocket.balance, '-3500')
+    })
+
+    it('should reject if the amount is higher than the other party will accept', async function () {
+      this.serverSocket.setMaxBalance(900)
+      let errored = false
+      try {
+        await this.clientSocket.pay(2000)
+      } catch (err) {
+        errored = true
+      }
+      assert.equal(errored, true)
```

Comment:
This could be done with assert.isRejected, and then you can also check the error message to make sure it's failing in the expected way